java/arcs/core/data/Ttl.kt

```diff
@@ -24,15 +26,33 @@ sealed class Ttl(count: Int, val isInfinite: Boolean = false) {
             "must be either positive count on infinite, " +
                 "but got count=$count and isInfinite=$isInfinite"
         }
+        if (!isInfinite) {
+            requireNotNull(time) {
+                "time cannot be null for non infinite ttl"
+            }
+        }
```

Comment:
Could be simplified into a single `require` check:

```kotlin
require((isInfinite && time == null) || (time != null && !isInfinite)) {
    "time must be provided if isInfinite == false"
}
```

Reply:
it's ok for `time` to be set even creating an `Infinite` ttl... otoh maybe I should just require it is always set?

Reply:
that could work

Reply:
removed the static `time` from ttl.